projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx

```diff
@@ -182,15 +177,54 @@ export default class DomEventHandler extends Component {
 		}
 	};
 
-	scrollOverlayToTop = event => {
-		// NOTE: the propertyName need to be aligned with the animation
-		if ( event?.propertyName !== 'opacity' ) {
-			return;
+	fixBodyScroll = () => {
+		if ( this.props.isVisible ) {
+			this.preventBodyScroll();
+			// This ensures the search input is visible on mobile devices.
+			// @see https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo
+			window?.scrollTo( 0, 0 );
+		} else if ( ! this.props.isVisible ) {
+			this.restoreBodyScroll();
 		}
-		// @see https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo
-		window?.scrollTo( 0, 0 );
 	};
 
+	/**
+	 * 1) When the overlay is open, we set body to fixed position.
+	 * 2) Body would be scrolled to top, so we need to set top to where the scroll position was.
+	 * 3) And we remember the body postition in `this.top`
+	 */
+	preventBodyScroll() {
+		this.top = parseInt( window.scrollY ) || 0;
+		/**
+		 * For logged-in user, there's a WP Admin Bar which is made sticky by adding `margin-top` to the document (the old way of `position: sticky;`).
+		 * So we need to fix the offset of scrollY for fixed positioned body.
+		 */
+		const scrollYOffset = document.documentElement?.scrollHeight - document.body?.scrollHeight || 0;
+		// Keep body at the same position when overlay is open.
+		document.body.style.top = `-${ this.top - scrollYOffset }px`;
+		// Make body in the center.
+		document.body.style.left = 0;
+		document.body.style.right = 0;
+		// Make body not scrollable.
+		document.body.style.position = 'fixed';
+	}
+
+	/**
+	 * 1) Unset body fixed postion
+	 * 2) Scroll back to the `this.top`
+	 * 3) Reset `this.top` to `0`
+	 */
+	restoreBodyScroll() {
+		// Restore body scroll.
+		document.body.style.top = '';
+		document.body.style.left = '';
+		document.body.style.right = '';
+		document.body.style.position = '';
+		// Restore body position.
+		this.top > 0 && window.scrollTo( 0, this.top );
+		this.top = 0;
```

Comment:
I'm currently seeing the following behavior when I try this in both desktop and mobile viewports:

1. Using the 2020 theme with a search input in the footer sidebar, open `/`.
2. Scroll all the way down to the footer and type a query into the input from (1).
3. The overlay opens as expected. The body scroll is affixed in the background, which is also expected.
4. Close the overlay.
5. The body abruptly scrolls to the very top and then smoothly scrolls back down to the position from (2).

To fix the abrupt scrolling in (5), we want to temporarily disable `smooth-scroll` on the `html` element like so:

```suggestion
		// Restore body position.
		const previousValue = document.querySelector( 'html' ).style[ 'scroll-behavior' ];
		document.querySelector( 'html' ).style[ 'scroll-behavior' ] = 'revert';
		this.top > 0 && window.scrollTo( 0, this.top );
		document.querySelector( 'html' ).style[ 'scroll-behavior' ] = previousValue ?? '';
		this.top = 0;
```

Reply:
Hi @jsnmoon  wondering if we should use ` document.documentElement.style.scrollBehavior` instead of ` document.querySelector( 'html' ).style[ 'scroll-behavior' ]` because style is an instance of `CSSStyleDeclaration` and `documentElement` looks a bit nicer 😄

Reply:
> Hi @jsnmoon wondering if we should use ` document.documentElement.style.scrollBehavior` instead of ` document.querySelector( 'html' ).style[ 'scroll-behavior' ]` because style is an instance of `CSSStyleDeclaration` and `documentElement` looks a bit nicer 😄

If that fixes the issue, then sure! I made my initial suggestion because Chrome's default stylesheet defines the `scroll-behavior` value on the root `html` element instead of the document.

Reply:
![image](https://user-images.githubusercontent.com/1425433/131419599-5dcc641c-d4e7-45ba-a055-c44c1f029d3a.png)
The fix doesn't work for twentytwenty as it sets an `important` body `position`.... 😭

Reply:
This is now fixed.

Reply:
`!important !important !no-really-this-is-more-important`